src/governance-classes.cpp

```diff
@@ -0,0 +1,671 @@
+    ParsePaymentSchedule(strAddresses, strAmounts);
+    
+    DBG( cout << "CSuperblock Constructor End" << endl; );
+}
+
+void 
+CSuperblock::
+ParsePaymentSchedule(std::string& strPaymentAddresses, std::string& strPaymentAmounts)
+{
+    // SPLIT UP ADDR/AMOUNT STRINGS AND PUT IN VECTORS
+
+    std::vector<std::string> vecParsed1;
+    std::vector<std::string> vecParsed2;
+    vecParsed1 = SplitBy(strPaymentAddresses, "|");
+    vecParsed2 = SplitBy(strPaymentAmounts, "|");
+
+    // IF THESE DONT MATCH, SOMETHING IS WRONG
+
+    if (vecParsed1.size() != vecParsed2.size()) {
+        std::ostringstream ostr;
+        ostr << "CSuperblock::ParsePaymentSchedule Mismatched payments and amounts";
+        throw std::runtime_error(ostr.str());
+    }
+
+    if (vecParsed1.size() == 0) {
+        std::ostringstream ostr;
+        ostr << "CSuperblock::ParsePaymentSchedule Error no payments";
+        throw std::runtime_error(ostr.str());
+    }
+
+    // LOOP THROUGH THE ADDRESSES/AMOUNTS AND CREATE PAYMENTS
+    /*
+      ADDRESSES = [ADDR1|2|3|4|5\6]
+      AMOUNTS = [AMOUNT1|2|3|4|5\6]
+    */
+
+    DBG( cout << "CSuperblock::ParsePaymentSchedule vecParsed1.size() = " << vecParsed1.size() << endl; );
+
+    for (int i = 0; i < (int)vecParsed1.size(); i++) {
+        CBitcoinAddress address(vecParsed1[i]);
+        if (!address.IsValid())  {
+            std::ostringstream ostr;
+            ostr << "CSuperblock::ParsePaymentSchedule Invalid Dash Address : " <<  vecParsed1[i];
+            throw std::runtime_error(ostr.str());
+        }
+
+        DBG( cout << "CSuperblock::ParsePaymentSchedule i = " << i 
+             <<  ", vecParsed2[i] = " << vecParsed2[i]
+             << endl; );
+
+        CAmount nAmount = ParsePaymentAmount(vecParsed2[i]);
+
+        DBG( cout << "CSuperblock::ParsePaymentSchedule: "
+             << "amount string = " << vecParsed2[i]
+             << ", nAmount = " << nAmount
+             << endl; );
+
+        CGovernancePayment payment(address, nAmount);
+        if(payment.IsValid()) {
+            vecPayments.push_back(payment);
+        }
+    }
+}
+
+
+/**
+*   Is Transaction Valid
+*
+*   - Does this transaction match the superblock?
+*/
+
+bool CSuperblock::IsValid(const CTransaction& txNew)
+{
+    // TODO : LOCK(cs);
+    // No reason for a lock here now since this method only accesses data
+    // internal to *this and since CSuperblock's are accessed only through
+    // shared pointers there's no way our object can get deleted while this
+    // code is running.
+
+    std::string strPayeesPossible = "";
+
+    // CONFIGURE SUPERBLOCK OUTPUTS 
+
+    int nPayments = CountPayments();    
+    for(int i = 0; i < nPayments; i++)
+    {
+        CGovernancePayment payment;
+        if(GetPayment(i, payment))
+        {
+            // SET COINBASE OUTPUT TO SUPERBLOCK SETTING
+
+            if(payment.script == txNew.vout[i].scriptPubKey && payment.nAmount == txNew.vout[i].nValue)
```

Comment:
imo if there is nothing to do when correct one is found, then this
```
+            if(payment.script == txNew.vout[i].scriptPubKey && payment.nAmount == txNew.vout[i].nValue)
+            {
+                // WE FOUND THE CORRECT SUPERBLOCK OUTPUT!
+            } else {
+                // MISMATCHED SUPERBLOCK OUTPUT!
```
could be replaced with
```
+            if(payment.script != txNew.vout[i].scriptPubKey || payment.nAmount != txNew.vout[i].nValue)
+            {
+                // MISMATCHED SUPERBLOCK OUTPUT!
```
to make it a bit more readable